test/unit/three_scale/redis_config_test.rb

```diff
@@ -21,14 +21,18 @@ class RedisConfigTest < ActiveSupport::TestCase
     end
 
     test 'sentinels' do
-      config = RedisConfig.new(url: 'redis://my-redis/1', sentinels: 'redis://:abc@127.0.0.1,localhost,redis://:passwd@external-redis,redis://localhost:1234')
+      username = 'user'
+      password = 'abc'
+      config = RedisConfig.new(url: 'redis://my-redis/1', sentinels: "redis://#{username}:#{password}@127.0.0.1,localhost,redis://external-redis,redis://localhost:1234")
       expected_sentinels = [
-        { host: '127.0.0.1', port: 26379, password: 'abc' },
+        { host: '127.0.0.1', port: 26379},
         { host: 'localhost', port: 26379 },
-        { host: 'external-redis', port: 26379, password: 'passwd' },
```

Comment:
means that password from URL for one sentinel will not be used for the rest? Or the other way around - credentials in URL of one sentinel will be used for all sentinels?

Reply:
This is a [change](https://github.com/redis/redis-rb/blob/master/CHANGELOG.md#508) in the `redis-rb` API after version 5. Now sentinels credentials must be sent through the `:sentinel_username` and `:sentinel_password` params ([docs](https://github.com/redis/redis-rb?tab=readme-ov-file#sentinel-support)). The credentials for all sentinels must be the same, see the Redis [docs](https://redis.io/docs/latest/operate/oss_and_stack/management/sentinel/#configuring-sentinel-instances-with-authentication)